### ISSUE TITLE
Problem about Multi-GPU-Train architecture

### ISSUE THREAD

**Issue by zsnoob** [ArrowLuo/CLIP4Clip#112]
In main_task_retrieval.py, fuction "train_epoch", we can see:
``` python
        if n_gpu > 1:
            loss = loss.mean()  # mean() to average on multi-gpu.
```
But in modeling.py, there is:
``` python
        if self.training:
            visual_output = allgather(visual_output, self.task_config)
            video_mask = allgather(video_mask, self.task_config)
            sequence_output = allgather(sequence_output, self.task_config)
            torch.distributed.barrier()
```
which means ALL GPUs' results get synchronized here. For instance, when my batch size is 128, i have 2 GPU to train the model and before the code above, every output's 0th dimension is 64, half of 128. And after this code block, they will be aggregated and output's size will be 128. So, when we get loss value in main_task, there is only a scalar tensor. Looks like there is no need to use "loss.mean()".
I have some print results:
 
![1700469492366](https://github.com/ArrowLuo/CLIP4Clip/assets/80535234/85960331-c355-458e-b4f0-f64bafbb4992)

``` python
        if n_gpu > 1:
            print(loss)
            print(loss.size())
            loss = loss.mean()  # mean() to average on multi-gpu.
            print(loss)
            print(loss.size())
```
